### ISSUE TITLE
Crash if no internet connection

### ISSUE THREAD

**Issue by abm-1980** [mvo5/synaptic#94]
Hello,

Recently, I have tried to install an application via Synaptic Package Manager running in PCLinuxOS which uses Trinity Desktop Environment out of the box.

I have overlooked the network connection which was suddenly shutdown by the system after the previous update and as I proceed
to installation of another package, the application immediately closed without prior notice. I have lately noticed that the network was
down and after reconnecting and reopening Synaptic, I was able to install my desired package again without problems.

I was able to reproduce the issue by disconnecting from the network and then installing any packages from Synaptic which I ran
at the command line to help trace the root cause. As the system had no internet again, Synaptic have closed again without prior notice and on the command line, I got "Segmentation Fault" which means the application has crashed for some reason if there's no internet and that's the only message printed on the terminal screen. Nothing else followed.

But based from what I've heard from others, clicking the "Reload" button doesn't have issues at all even if there's no internet connection as the application notifies via pop-up message that it cannot download the package and it continues to run upon dismissing the said pop-up.

In summary, these were the test cases tried:

1. No internet connection and then proceed to install package, Synaptic crashed due to segmentation fault.
2. No internet connection and then click reload button prior to installation, Synaptic didn't crashed as it continued to run with pop-up message that it cannot download the package.

Case 2 is okay as expected but case 1 is not supposed to happen as both requires internet connection.

Case 1 is reproducible on my laptop by simply disconnecting from the network and then proceed to installation of any packages via Synaptic. Even in this case, the application is supposed to exhibit the same behavior like in case 2.

If there's anything else you need, just let me know. I'll look forward to hearing from you soon.
Thank you.

**Comment by abm-1980** [comment]
Sorry, I forgot to tell you that the version of Synaptic with issue is 0.57.11

**Comment by abm-1980** [comment]
I have tried to search through Google on how to update Synaptic Package Manager via apt-get tool but I haven't found yet the answer I need.

What I did now was, I updated the system with these commands:

apt-get update
apt-get dist-upgrade

Upon completion and reboot, the same version of the said application is still installed.
Any idea on how to upgrade to the latest version of Synaptic Package Manager?

I believe the issue I have mentioned yesterday is already resolved in the latest release.
Please advise.

Thank you.
